src/HDF5.jl

```diff
@@ -579,24 +580,37 @@ There are many keyword properties that can be set. Below are a few select keywor
 * `filters` - `AbstractVector{<: Filters.Filter}` describing the order of the filters to apply to the data. See [`Filters`](@ref)
 * `external` - `Tuple{AbstractString, Intger, Integer}` `(filepath, offset, filesize)` External dataset file location, data offset, and file size. See [`API.h5p_set_external`](@ref).
 
+Additionally, the initial create, transfer, and access properties can be provided as a keyword:
+* `dcpl` - [`DatasetCreateProperties`](@ref)
+* `dxpl` - [`DatasetTransferProperties`](@ref)
+* `dapl` - [`DatasetAccessProperties`](@ref)
+
 See also
 * [`H5P`](@ref H5P)
-* [`DatasetCreateProperties`](@ref)
-* [`DatasetTransferProperties`](@ref)
-* [`DatasetAccessProperties`](@ref)
 """
-function create_dataset(parent::Union{File,Group}, path::AbstractString, dtype::Datatype, dspace::Dataspace; pv...)
-    haskey(parent, path) && error("cannot create dataset: object \"", path, "\" already exists at ", name(parent))
-    dcpl = DatasetCreateProperties()
-    dxpl = DatasetTransferProperties()
-    dapl = DatasetAccessProperties()
+function create_dataset(
+    parent::Union{File,Group},
+    path::Union{AbstractString, Nothing},
+    dtype::Datatype,
+    dspace::Dataspace;
+    dcpl::DatasetCreateProperties = DatasetCreateProperties(),
+    dxpl::DatasetTransferProperties = DatasetTransferProperties(),
+    dapl::DatasetAccessProperties = DatasetAccessProperties(),
+    pv...
+)
+    path !== nothing && haskey(parent, path) && error("cannot create dataset: object \"", path, "\" already exists at ", name(parent))
```

Comment:
worth to test this as well?

Reply:
Could you clarify what you would like to test? I already test passing `nothing` to the function.

The code for checking for an existing dataset already existed before this PR. Is there currently not a test for this?

Reply:
I think we already test the error for an existing dataset:
https://github.com/JuliaIO/HDF5.jl/blob/704d9e8c32b3b5d1239254d0155d7e14cad0c713/test/plain.jl#L1362-L1367

Reply:
Nevermind. When I read this, I thought you added this whole line, but that's not right heh